test/seqan3-test.cmake

```diff
@@ -23,6 +29,30 @@ option (SEQAN3_TEST_BUILD_OFFLINE "Skip the update step of external projects." O
 # have to be adapted or the option deactivated.
 option (SEQAN3_BENCHMARK_ALIGN_LOOPS "Pass -falign-loops=32 to the benchmark builds." ON)
 
+option (SEQAN3_WITH_SEQAN2 "Build tests with SeqAn2." OFF)
+# Will be on if environment variable CI is set.
+# Can be toggled off if initially set to on.
+# Cannot be toggled on if initially set to off.
+cmake_dependent_option (SEQAN3_WITH_SEQAN2_CI "Build tests with SeqAn2." ON "DEFINED ENV{CI}" OFF)
+
+if (SEQAN3_WITH_SEQAN2 OR SEQAN3_WITH_SEQAN2_CI)
```

Comment:
This seems a little over complicated. would vote for just the SEQAN3_WITH_SEQAN2 flag and the .github/workflow/xyz.yml file just have to set the flag on the command line when cmake is being called.